server/client.go

```diff
@@ -3125,6 +3117,16 @@ func (c *client) deliverMsg(sub *subscription, subject, reply, mh, msg []byte, g
 	return true
 }
 
+func (c *client) serveUpdate(client *Client, sub *subscription) {
```

Comment:
I would rename this to make it a server receiver since when we call above, you already have `srv`. Also, maybe a rename to `func (s *Server) updateRemoteSubscription(c *client, sub *subscription)`?